### PR TITLE
Hide columns in unused entities on mobile

### PR DESCRIPTION
**Pull request by bramkragten**
Schermafbeelding 2019-09-14 om 18 36 34

### CODE REVIEW

**Comment by bramkragten** [comment]
Using `paper-icon-item` might be even better as on mobile we want to use the height and not the width to show almost the same info.

Just can't get the active style background to hide.

Schermafbeelding 2019-09-15 om 11 48 15

**Comment by SeanPM5** [comment]
This revision is a million times more mobile-friendly than the first one, nice! 👍

One small suggestion: maybe hide the floating plus icon until a checkbox is tapped first? 

I assume it can’t be used unless an entity is selected, so probably no reason to show it by default?

File: src/panels/lovelace/editor/unused-entities/hui-unused-entities.ts [hunk 1]
```diff
@@ -45,40 +50,62 @@ export class HuiUnusedEntities extends LitElement {
...
-        <ha-relative-time
-          .hass=${this.hass!}
-          .datetime=${lastChanged}
-        ></ha-relative-time>
-      `,
-    },
-  };
+  private _columns = memoizeOne((narrow: boolean) =>
+    narrow
+      ? {
```
Comment (balloob) [inline src/panels/lovelace/editor/unused-entities/hui-unused-entities.ts]: This works but we're duplicating `entity` now. Can we do something like this?

```ts
const columns = {
  entity: { … }
}

if (narrow) {
  return columns;
}

columns.entity_id = {…}
columns.domain = {…}
return columns;
```

File: src/panels/lovelace/editor/unused-entities/hui-unused-entities.ts [hunk 2]
```diff
@@ -46,42 +51,53 @@ export class HuiUnusedEntities extends LitElement {
...
+            ${stateObj.friendly_name}
+          </div>
+        `,
+      },
+    };
+
+    return narrow
+      ? columns
+      : {
+          ...columns,
```
Comment (balloob) [inline src/panels/lovelace/editor/unused-entities/hui-unused-entities.ts]: Now you're copying an object but we own the old object and can mutate it.

**Review by balloob — APPROVED** [review]
ok to merge when final comment addressed

**Pull request merged**